pkg/controllers/disruption/controller.go

```diff
@@ -127,7 +108,7 @@ func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconc
 	// Karpenter taints nodes with a karpenter.sh/disruption taint as part of the disruption process
 	// while it progresses in memory. If Karpenter restarts during a disruption action, some nodes can be left tainted.
 	// Idempotently remove this taint from candidates before continuing.
-	if err := c.requireNoScheduleTaint(ctx, false, c.cluster.Nodes()...); err != nil {
+	if err := c.Queue.RequireNoScheduleTaint(ctx, false, c.cluster.Nodes()...); err != nil {
```

Comment:
Since the queue needs to be able to un-cordon on a failed action, and the disruption package depends on the orchestration package, it's either duplicate two utils or use the member function like this.